src/Native/CpuMathNative/Sse.cpp

```diff
@@ -194,7 +194,7 @@ EXPORT_API(void) MatMul(_In_ const float * pmat, _In_ const float * psrc, _Inout
             if (length > 3)
             {
                 remainder = length % 4;
-                while(pSrcCurrent < pSrcEnd)
+                while(pSrcCurrent + 4 <= pSrcEnd)
```

Comment:
>pSrcCurrent [](start = 22, length = 11)

A) can you run auto format on this file, I can see bunch of empty spaces after line.
B) more importantly, can you add comments to code? I assume you figured out what is going on here, would be nice to have some comments on top of if statements, overwise I would probably spend half of hour by just looking into all this conditions before I get to code you changed.